dough-skins/src/main/java/io/github/bakedlibs/dough/skins/nms/PlayerHeadAdapterPaper.java

```diff
@@ -0,0 +1,35 @@
+package io.github.bakedlibs.dough.skins.nms;
+
+import com.destroystokyo.paper.profile.PlayerProfile;
+import com.destroystokyo.paper.profile.ProfileProperty;
+import com.mojang.authlib.GameProfile;
+import com.mojang.authlib.properties.Property;
+import org.bukkit.Bukkit;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Skull;
+
+import javax.annotation.ParametersAreNonnullByDefault;
+
+public class PlayerHeadAdapterPaper implements PlayerHeadAdapter {
+
+    @Override
+    @ParametersAreNonnullByDefault
+    public void setGameProfile(Block block, GameProfile profile, boolean sendBlockUpdate) {
+        BlockState state = block.getState();
+        if (!(state instanceof Skull)) return;
+
+        Skull skull = (Skull) state;
+
+        Property property = profile.getProperties().get("textures").iterator().next();
+
+        PlayerProfile paperPlayerProfile = Bukkit.createProfile(profile.getId(), profile.getName());
+        paperPlayerProfile.setProperty(new ProfileProperty(property.name(), property.value(), property.signature()));
```

Comment:
RecordJava started using records in Java 16, while Authlib didn't use records until much later. As of the last version of Paper 1.18.2 that depended on Authlib, Property was still a class. Therefore, you might need to use reflection to call the getter method of an older version of Property.